reframe/core/schedulers/flux.py

```diff
@@ -74,6 +76,12 @@ def submit(self, job):
         job._submit_time = time.time()
         job._flux_future = flux_future
 
+    def cancel(self, job):
+        # Job cannot cancel once running or completed
+        if not job._flux_future.cancel():
+            # This will raise JobException with event=cancel
+            flux.job.cancel(flux.Flux(), job._flux_future.jobid())
```

Comment:
If this is gonna raise, it's better if you wrap it and reraise it as a `JobError`. Reframe's runtime can cancel jobs if a timeout is reached or if the user aborts the execution. If the raised error is a `JobError` and generally any other soft error, reframe will just print its message without a stack trace, otherwise the stack trace will be printed. Generally, you can enable stack traces also for soft errors, such as `JobError` if you increase verbosity, but users wouldn't want to see a full stack trace by default if the job was not cancelled.

```python
try:
    flux.job.cancel(flux.Flux(), job._flux_future.jobid())
except flux.job.JobException as err:
    raise JobError('some message') from err
```

Reply:
The second command (the one you targeted) won't raise directly - as I understand it, it's a future so will only raise when we retrieve the result (in the poll function). So I don't think this is necessary here - we can bring in the flux maintainers after a first round of review to double check this.

Reply:
If that's the case, then I think that's the desired behaviour. I was confused a bit the comment and thought that this would raise.

Reply:
Sorry bad comment, I’ll tweak it for next round of pushes to make it clear that happens later!